QuantConnect.CoinbaseBrokerage/CoinbaseBrokerage.Messaging.cs

```diff
@@ -297,48 +357,58 @@ private void EmitQuoteTick(Symbol symbol, decimal bidPrice, decimal bidSize, dec
-            var pendingSymbols = new List<Symbol>();
-            foreach (var item in fullList)
-            {
-                if (_symbolMapper.IsKnownLeanSymbol(item))
-                {
-                    pendingSymbols.Add(item);
-                }
-                else if (item.SecurityType == SecurityType.Crypto)
-                {
-                    Log.Error($"{nameof(CoinbaseBrokerage)}:{nameof(Subscribe)}: Unknown symbol: {item.Value}");
-                }
-            }
+            SubscribeSymbolsOnDataChannels(symbols.ToList());
 
-            var products = pendingSymbols.Select(symbol => _symbolMapper.GetBrokerageSymbol(symbol)).ToList();
-
-            if (products.Count == 0)
-            {
                 return true;
             }
 
-            foreach (var channel in CoinbaseWebSocketChannels.WebSocketChannelList)
+        /// <summary>
+        /// Ends current subscriptions
+        /// </summary>
+        public bool Unsubscribe(IEnumerable<Symbol> leanSymbols)
+        {
+            SubscribeSymbolsOnDataChannels(leanSymbols.ToList(), WebSocketSubscriptionType.Unsubscribe);
+
+            return true;
+        }
+
+        /// <summary>
+        /// Restores data subscriptions existing
+        /// </summary>
+        private void RestoreDataSubscriptions()
+        {
+            List<Symbol> subscribedSymbols;
+            lock (_synchronizationContext)
             {
-                ManageChannelSubscription(WebSocketSubscriptionType.Subscribe, channel, products);
+                subscribedSymbols = SubscriptionManager.GetSubscribedSymbols().ToList();
             }
 
-            return true;
+            SubscribeSymbolsOnDataChannels(subscribedSymbols);
         }
 
         /// <summary>
-        /// Ends current subscriptions
+        /// Subscribes to real-time data channels for the provided list of symbols.
         /// </summary>
-        public bool Unsubscribe(IEnumerable<Symbol> leanSymbols)
+        /// <param name="symbols">The list of symbols to subscribe to.</param>
+        /// <remarks>
+        /// This method subscribes to WebSocket channels for each provided symbol, converting them to brokerage symbols using
+        /// the symbol mapper. It then iterates through the available WebSocket channels and manages the subscription by
+        /// invoking the <see cref="ManageChannelSubscription"/> method with the appropriate parameters.
+        /// </remarks>
+        /// <seealso cref="ManageChannelSubscription"/>
+        private void SubscribeSymbolsOnDataChannels(List<Symbol> symbols, WebSocketSubscriptionType subscriptionType = WebSocketSubscriptionType.Subscribe)
         {
-            var brokerageSymbols = leanSymbols.Select(symbol => _symbolMapper.GetBrokerageSymbol(symbol)).ToList();
+            var products = symbols.Select(symbol => _symbolMapper.GetBrokerageSymbol(symbol)).ToList();
+
+            if (products.Count == 0)
+        {
```

Comment:
nit - tab/spacing issues. In VS Ctrl+K+D should to the trick

Reply:
Applied format the entire document.